lib/location/method_gnss.c

```diff
@@ -109,6 +109,54 @@ static void method_gnss_pgps_ext_work_fn(struct k_work *item);
 
 static int fixes_remaining;
 
+static inline int check_modem_supports_gnss(void)
```

Comment:
I don't remember if it possible for neither MQTT nor REST be defined. If so, compiler warning about unused function might pop up? Maybe consider putting this function definition inside ifdef MQTT || MQTT.

Reply:
`check_modem_supports_gnss` is always referenced by `method_gnss_location_get`, so I don't think that will be an issue, but `check_modem_supports_lte` is conditionally referenced. However even with `MQTT || REST` guaranteed, I believe it is possible for the function to not be referenced. I also worry about the potential upkeep implications down the line.

Fortunately, it seems that my use of the inline keyword solves this issue; No warnings are generated when I create a static inline function `check_modem_supports_funny_business` with no references to it. Not sure what we would do if inlining were not an option, but it seems workable in this case.

Does just moving forward with the functions inlined seem reasonable to you, @hiltunent ?

Reply:
`check_modem_supports_gnss()` is only called from a single place, so there making the function inline doesn't probably make any difference. But `check_modem_supports_lte()` on the other hand is called from three places and having the function inline unnecessarily increases the flash consumption, because the same code will be on flash three times. Because of that flagging would be preferable.

Reply:
While having `check_modem_supports_gnss()` as inline is technically fine, I'm not sure if we tend to have this type of functions with few if-statements as inline normally. If somebody starts using it in another place, is (s)he going to remove the inline part then? I guess that might easily be missed.

Reply:
I am not 100% certain flagging is preferable. I agree that increasing the flash usage is non-ideal. I also agree that relying on inlining to prevent compiler warnings seems wrong.

However, I don't think flagging is a good solution either. The exact inclusion conditions for `check_modem_supports_lte` is `(AGPS && !AGPS_EXTERNAL && (MQTT || REST)) || (PGPS && !MQTT)` due to the various conditions under which the functions which rely upon it may be enabled. I worry that this logic will be difficult for a reader to follow and that it will be difficult to ensure the logic is updated if the inclusion conditions ever change.

After some thought, the best solution, I think would be to move the functions to `location_utils.c` and give them header definitions, leaving them non-inline.

I will switch to this approach.

Reply:
@trantanen I think I have resolved the issue, do you mind taking a look at my latest approach and checking whether it makes sense to you?

Reply:
Yep, this is fine for me.